contracts/Company/Extras/GranularTransferController.sol

```diff
@@ -0,0 +1,187 @@
+pragma solidity 0.4.26;
+
+import "../PlaceholderEquityTokenController.sol";
+
+
+contract GranularTransferController is
+    PlaceholderEquityTokenController
+{
+    ////////////////////////
+    // Type declarations
+    ////////////////////////
+
+    /// represents right to force transfer by the company
+    struct ForcedTransfer {
+        // to account
+        address to;
+        // amount
+        uint256 amount;
+    }
+
+    ////////////////////////
+    // Mutable state
+    ////////////////////////
+
+    // keeps list of frozen addresses that cannot transfer or receive transfers
+    mapping(address => bool) private _frozenAddresses;
+
+    // list of available forced transfers (from => ForcedTransfer)
+    mapping(address => ForcedTransfer) private _forcedTransfers;
+
+    ////////////////////////
+    // Events
+    ////////////////////////
+
+    event LogForcedTransferEnabled(
+        address from,
+        address to,
+        uint256 amount
+    );
+
+    event LogForcedTransferExecuted(
+        address token,
+        address from,
+        address to,
+        uint256 amount
+    );
+
+    event LogTokenHolderAccountFrozen(
+        address owner
+    );
+
+    event LogTokenHolderAccountUnfrozen(
+        address owner
+    );
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    constructor(
+        Universe universe,
+        address companyLegalRep
+    )
+        public
+        PlaceholderEquityTokenController(universe, companyLegalRep)
+    {}
+
+    //
+    // Implements ITokenController
+    //
+
+    function onTransfer(address broker, address from, address to, uint256 amount)
+        public
+        constant
+        returns (bool allow)
+    {
+        // ask base controller if transfers are enables
+        allow = PlaceholderEquityTokenController.onTransfer(broker, from, to, amount);
+        GovState s = state();
+        // allow forced transfer only by the token controller itself
+        if ((s == GovState.Funded || s == GovState.Closing) && broker == address(this)) {
+            ForcedTransfer storage t = _forcedTransfers[from];
+            // check if forced transfer matches the actual transfer
+            if (t.amount > 0) {
+                // we allow because we checked the following
+                //  (1) the broker is the company
+                //  (2) there's forced transfer defined for from address
+                //  (3) there's a match for to and amount
+                allow = t.amount == amount && t.to == to;
```

Comment:
shouldnt this be:

```allow = allow && (t.amount == amount && t.to == to);```

?

I dont prefectely understand the conditions including GovState but even if they only pass if allow is already true it looks very implicit

Reply:
but you are right - it does not make sense to obtain `allow` from base class if the whole condition does not need it! it should be done when this is really needed.

saves gas. will fix it.